### PR TITLE
style: add responsive-header, add globalstyles, fix margin expenses

### PR DESCRIPTION
**Pull request by HarryPotter-4000**
### Description

Please provide a brief description of the changes you have made in your PR.

### Type of change

- [ ] Feature (introduces new functionality)
- [x] Bugfix
- [x] Configuration
- [x] Refactoring (doesn't change functionality, only the code)
- [x] Style changes (doesn't affect functionality, only the look of the components)

### PR Checklist

- [x] Synced up with latest from the main branch/Code is linted
- [x] Description of the changes has been added (Screenshots/Video)
- [x] Chakra components/constants/hooks have been used wherever possible

> If not explain here

- [x] Commit messages follow [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#summary)
- [x] Console has been checked for warnings

#### Screenshots/Video before change

exp-b

Add screenshots from the [current deployed state](https://sandbox-money-app.herokuapp.com/) of the changed components here.

#### Screenshots/Video after change

Fix margin top:

exp-b
Desktop:
resp-desk
Tablet:
resp-mob

- [x] Commit messages follow [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#summary)
- [x] Console has been checked for warnings

#### Screenshots/Video before change

exp-b

Add screenshots from the [current deployed state](https://sandbox-money-app.herokuapp.com/) of the changed components here.

#### Screenshots/Video after change

Fix margin top:

exp-b
Desktop:
resp-desk
Tablet:
resp-mob

### CODE REVIEW

**Comment by HarryPotter-4000** [comment]
Mikalai suggests adding a maximum width in global styles, for not to write them every time for each page. But they do not work well with the modal, so I removed them in second commit. It would be convenient for everyone if it worked.

File: src/components/Header/NavDrawer.jsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          borderStyle={'solid'}
+          borderColor={iconsThemeColor}
+          bgColor={headerBgThemeColor}
+          color={iconsThemeColor}
+          ref={btnRef}
+          size={'sm'}
+          icon={isOpen ? <CloseIcon /> : <HamburgerIcon fontSize={'20px'} />}
+          aria-label={'Open Menu'}
+          display={{ base: 'block', lg: 'none' }}
+          onClick={isOpen ? onClose : onOpen}
```
Comment (kalashnikau) [inline src/components/Header/NavDrawer.jsx:41]:
```suggestion
          onClick={onToggle}
```

useDisclosure has `onToggle` method which you can use to achieve same result

File: src/components/Header/NavDrawer.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  IconButton,
+  useColorModeValue,
+  useDisclosure
+} from '@chakra-ui/react';
+
+import { Navbar } from './Navbar';
+import { UserName } from './UserName';
+
+export const NavDrawer = () => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
```
Comment (kalashnikau) [inline src/components/Header/NavDrawer.jsx:20]:
```suggestion
  const { isOpen, onOpen, onClose, onToggle } = useDisclosure();
```

File: src/theme/theme.js [hunk 3]
```diff
@@ -18,6 +18,11 @@ const config = {
 };
 
 export const GlobalStyles = css`
+  // body {
+  //   max-width: 1080px;
+  //   margin: 0 auto;
+  //   float: none;
+  // }
```
Comment (kalashnikau) [inline src/theme/theme.js:25]: remove or uncomment

File: src/components/Header/UserName.jsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import { Flex, Text, useColorModeValue } from '@chakra-ui/react';
+
+import loginAction from '@/api/AuthProvider';
+
+export const UserName = () => {
+  const { data } = useQuery(['user'], () => loginAction());
+  const headerTextColor = useColorModeValue('teal.900', 'orange.300');
+
+  return (
+    <Flex direction="column" align={{ base: '"flex-start"', xl: 'flex-end' }}>
```
Comment (kalashnikau) [inline src/components/Header/UserName.jsx:12]:
```suggestion
    <Flex direction="column" align={{ base: 'flex-start', xl: 'flex-end' }}>
```

File: src/components/Header/UserName.jsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { useQuery } from 'react-query';
+import { Flex, Text, useColorModeValue } from '@chakra-ui/react';
+
```
Comment (kalashnikau) [inline src/components/Header/UserName.jsx:4]: remove extra space

File: src/components/Header/Notification.jsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { BellIcon } from '@chakra-ui/icons';
+import { IconButton, useColorModeValue } from '@chakra-ui/react';
+import i18next from 'i18next';
+
+export const Notification = () => {
+  const iconsThemeColor = useColorModeValue('teal.900', 'orange.300');
+  const iconsHoverThemeColor = useColorModeValue('teal.900', 'orange.300');
```
Comment (kalashnikau) [inline src/components/Header/Notification.jsx:7]: can we somehow have that on theme level? cause those lines duplicated almost in every component

File: src/components/Header/Header.jsx [hunk 7]
```diff
@@ -1,86 +1,55 @@
...
+      boxShadow="headerShadow"
+      color={headerTextColor}
     >
       <Flex
         align="center"
         justify="space-between"
-        maxW="1200px"
-        w="100%"
+        maxWidth="1200px"
+        w={'100%'}
```
Comment (kalashnikau) [inline src/components/Header/Header.jsx:29]:
```suggestion
        w="100%"
```

File: src/components/Header/NavDrawer.jsx [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+      <Flex flexGrow={{ sm: '2', md: '2', lg: '0', xl: '0' }}>
+        <IconButton
+          marginLeft="auto"
+          borderWidth={'2px'}
+          borderStyle={'solid'}
+          borderColor={iconsThemeColor}
+          bgColor={headerBgThemeColor}
+          color={iconsThemeColor}
+          ref={btnRef}
+          size={'sm'}
```
Comment (kalashnikau) [inline src/components/Header/NavDrawer.jsx:37]:
```suggestion
          size="sm"
```

File: src/components/Header/NavDrawer.jsx [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+  const iconsThemeColor = useColorModeValue('teal.900', 'orange.300');
+  const userMenuBgThemeColor = useColorModeValue('orange.50', 'teal.800');
+
+  return (
+    <>
+      <Flex flexGrow={{ sm: '2', md: '2', lg: '0', xl: '0' }}>
+        <IconButton
+          marginLeft="auto"
+          borderWidth={'2px'}
+          borderStyle={'solid'}
```
Comment (kalashnikau) [inline src/components/Header/NavDrawer.jsx:32]:
```suggestion
          borderWidth="2px"
          borderStyle="solid"
```

File: src/components/Header/NavDrawer.jsx [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+          borderColor={iconsThemeColor}
+          bgColor={headerBgThemeColor}
+          color={iconsThemeColor}
+          ref={btnRef}
+          size={'sm'}
+          icon={isOpen ? <CloseIcon /> : <HamburgerIcon fontSize={'20px'} />}
+          aria-label={'Open Menu'}
```
Comment (kalashnikau) [inline src/components/Header/NavDrawer.jsx:39]: need to be localized

File: src/components/Header/NavDrawer.jsx [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+  DrawerCloseButton,
+  DrawerContent,
+  DrawerHeader,
+  DrawerOverlay,
+  Flex,
+  IconButton,
+  useColorModeValue,
+  useDisclosure
+} from '@chakra-ui/react';
+
```
Comment (kalashnikau) [inline src/components/Header/NavDrawer.jsx:15]: any specific reason for having extra space?

Comment (HarryPotter-4000) [reply src/components/Header/NavDrawer.jsx:15]: It's from eslint: "simple-import-sort/imports"
eslint

File: src/components/Header/UserName.jsx (hunk 5 above)
Comment (HarryPotter-4000) [reply src/components/Header/UserName.jsx:4]: It's from eslint: "simple-import-sort/imports"
eslint

File: src/theme/theme.js (hunk 3 above)
Comment (HarryPotter-4000) [reply src/theme/theme.js:25]: done

File: src/components/Header/UserName.jsx (hunk 4 above)
Comment (HarryPotter-4000) [reply src/components/Header/UserName.jsx:12]: done

File: src/components/Header/Notification.jsx (hunk 6 above)
Comment (HarryPotter-4000) [reply src/components/Header/Notification.jsx:7]: done

File: src/components/Header/NavDrawer.jsx (hunk 8 above)
Comment (HarryPotter-4000) [reply src/components/Header/NavDrawer.jsx:37]: done

File: src/components/Header/NavDrawer.jsx (hunk 9 above)
Comment (HarryPotter-4000) [reply src/components/Header/NavDrawer.jsx:32]: done

File: src/components/Header/NavDrawer.jsx (hunk 10 above)
Comment (HarryPotter-4000) [reply src/components/Header/NavDrawer.jsx:39]: done

File: src/components/Header/Header.jsx (hunk 7 above)
Comment (HarryPotter-4000) [reply src/components/Header/Header.jsx:29]: done

File: src/components/Header/NavDrawer.jsx (hunk 2 above)
Comment (HarryPotter-4000) [reply src/components/Header/NavDrawer.jsx:20]: done

File: src/components/Header/NavDrawer.jsx (hunk 1 above)
Comment (HarryPotter-4000) [reply src/components/Header/NavDrawer.jsx:41]: done

**Review by kalashnikau — APPROVED** [review]

**Review by Yulatun — APPROVED** [review]

File: src/components/Header/Header.jsx [hunk 12]
```diff
@@ -1,86 +1,55 @@
...
-      boxShadow="base"
+      boxShadow="headerShadow"
+      color={headerTextColor}
     >
       <Flex
         align="center"
         justify="space-between"
-        maxW="1200px"
-        w="100%"
+        maxWidth="1200px"
```
Comment (lazzzaroni) [inline src/components/Header/Header.jsx:28]:
```suggestion
        maxW="container.xl"
```

I suggest to use here Chakra's default value for max-width just to be more consistent.
Basically, it contains '1280px' value in it and we already have an access to that variable with Chakra's predefined sizes.

https://chakra-ui.com/docs/styled-system/theme#sizes

And we can use `maxW` shorthand here. I don't understand why you change it)

**Pull request merged**